src/Application/FileProcessor/PhpFileProcessor.php

```diff
@@ -83,6 +88,14 @@ public function process(File $file, Configuration $configuration): void
                 $this->printFile($file, $configuration);
             }, ApplicationPhase::PRINT());
         } while ($file->hasChanged());
+
+        // return json here
+        $fileDiff = $file->getFileDiff();
```

Comment:
@TomasVotruba it seems cause result `null` while it can have the diff, eg:

```bash
bin/rector process src/Util -c build/config/config-downgrade.php --clear-cache  -vvv
```

It shows ` [OK] Rector is done!` without diff:

```bash
 [OK] Rector is done!                                                                                                   
```


it's `$fileDiff` result is `null`, while checking the result, it shows:

```diff
$ git diff src/Util/

+++ b/src/Util/StaticRectorStrings.php
@@ -53,7 +53,7 @@ final class StaticRectorStrings
     public static function removePrefixes(string $value, array $prefixesToRemove): string
     {
         foreach ($prefixesToRemove as $prefixToRemove) {
-            if (\str_starts_with($value, $prefixToRemove)) {
+            if (strncmp($value, $prefixToRemove, strlen($prefixToRemove)) === 0) {
                 $value = Strings::substring($value, Strings::length($prefixToRemove));
             }
         }
@@ -67,7 +67,7 @@ final class StaticRectorStrings
     public static function removeSuffixes(string $value, array $suffixesToRemove): string
     {
         foreach ($suffixesToRemove as $suffixToRemove) {
-            if (\str_ends_with($value, $suffixToRemove)) {
+            if (substr_compare($value, $suffixToRemove, -strlen($suffixToRemove)) === 0) {
                 $value = Strings::substring($value, 0, -Strings::length($suffixToRemove));
             }
         }
```

debug the `FileDiffFileDecorator`, the diff shows:

```javascript
[print] src/Util/StaticRectorStrings.php
Rector\Core\ValueObject\Reporting\FileDiff #25533
   relativeFilePath: 'src/Util/StaticRectorStrings.php'
   diff: string
   |  '--- Original\n
   |   +++ New\n
   |   @@ -53,7 +53,7 @@\n
   |        public static function removePrefixes(string $value, array $prefixesToRemove): string\n
   |        {\n
   |            fore ...         }\n'
   diffConsoleFormatted: string
   |  '<comment>    ---------- begin diff ----------</comment>\n
   |   <fg=cyan>@@ @@</fg=cyan>\n
   |        public static function removePrefixes(string $value, array $pref ... /comment>\n'
   rectorsWithLineChanges: array (2)
   |  0 => Rector\ChangesReporting\ValueObject\RectorWithLineChange #25355
   |  |  rectorClass: 'Rector\DowngradePhp80\Rector\FuncCall\DowngradeStrStartsWithRector'
   |  |  line: 56
   |  1 => Rector\ChangesReporting\ValueObject\RectorWithLineChange #25457
   |  |  rectorClass: 'Rector\DowngradePhp80\Rector\FuncCall\DowngradeStrEndsWithRector'
   |  |  line: 70
```